lib/shopping_cart.rb

```diff
@@ -0,0 +1,43 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize (name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {name: name, capacity: capacity.delete("^0-9")}
+  end
+
+  def total_number_of_products
+    total = []
+    @products.each do |product|
+      total << product.quantity.to_i
+    end
+    total.sum
+  end
+
+  def is_full?
+    if total_number_of_products > @capacity.to_i
+      true
+    else false
+    end
+  end
+
+  # def products_by_category(x)
+  # end
+  #   @products.each do |product|
+  #     if product.category == x
+  #       puts product
+  #     end
+  #   end
+  # end
+    # @products.where(@product.category: x)
```

Comment:
You may have been `putsing` to the terminal for testing, but try and stay away from that. I think you're close here. If the product's category is equal to x, then you want to store it into a collection to later be returned. I think teh only thing you're missing here is a placeholder to be able to shovel products into that match the category passed in.